tests/test_geography.py

```diff
@@ -5,59 +5,117 @@
+    line = spherely.linestring(points)
     assert line.dimensions == 1
     assert line.nshape == 1
     assert repr(line).startswith("LINESTRING (5 50")
 
 
+@pytest.mark.parametrize(
+    "lines",
+    [
+        [[(5, 50), (6, 51)], [(15, 60), (16, 61)]],
+        [
+            [spherely.point(5, 50), spherely.point(6, 51)],
+            [spherely.point(15, 60), spherely.point(16, 61)],
+        ],
+        [
+            spherely.linestring([(5, 50), (6, 51)]),
+            spherely.linestring([(15, 60), (16, 61)]),
+        ],
+    ],
+)
+def test_multilinestring(lines) -> None:
+    multiline = spherely.multilinestring(lines)
+    assert multiline.dimensions == 1
+    assert multiline.nshape == 2
+    assert repr(multiline).startswith("MULTILINESTRING ((5 50")
+
+
 @pytest.mark.parametrize(
     "coords",
     [
-        [(0, 0), (0, 2), (2, 2), (2, 0)],
+        [(0, 0), (2, 0), (2, 2), (0, 2)],
         [
-            spherely.Point(0, 0),
-            spherely.Point(0, 2),
-            spherely.Point(2, 2),
-            spherely.Point(2, 0),
+            spherely.point(0, 0),
+            spherely.point(2, 0),
+            spherely.point(2, 2),
+            spherely.point(0, 2),
         ],
     ],
 )
 def test_polygon(coords) -> None:
-    poly = spherely.Polygon(coords)
+    poly = spherely.polygon(coords)
     assert poly.dimensions == 2
     assert poly.nshape == 1
     assert repr(poly).startswith("POLYGON ((0 0")
 
 
+@pytest.mark.parametrize(
+    "coords",
+    [
+        [(0, 0), (2, 0), (2, 2), (0, 2)],
+        [(0, 0), (2, 0), (2, 2), (0, 2), (0, 0)],
+    ],
+)
+def test_polygon_closing(coords) -> None:
+    # support both manual and automated closing
+    ring = spherely.polygon(coords)
+    assert repr(ring).startswith("POLYGON ((0 0")
+    assert repr(ring).endswith("0 0))")
+
+
 def test_polygon_error() -> None:
-    with pytest.raises(ValueError, match="loop is not valid.*duplicate vertex.*"):
-        # polygon vertices should be open (duplicate vertex error)
-        spherely.Polygon([(0, 0), (0, 2), (2, 0), (0, 0)])
+    with pytest.raises(ValueError, match="polygon is not valid.*duplicate vertex.*"):
+        spherely.polygon([(0, 0), (0, 2), (0, 2), (2, 0)])
 
-    with pytest.raises(ValueError, match="loop is not valid.*at least 3 vertices.*"):
-        spherely.Polygon([(0, 0), (0, 2)])
+    with pytest.raises(ValueError, match="polygon is not valid.*at least 3 vertices.*"):
+        spherely.polygon([(0, 0), (0, 2)])
+
+    with pytest.raises(ValueError, match="polygon is not valid.*Edge.*crosses.*"):
+        spherely.polygon([(0, 0), (2, 0), (1, 2), (1, -2)])
+
+    with pytest.raises(ValueError, match="polygon is not valid.*crosses.*"):
+        # shell/hole rings are crossing each other
+        spherely.polygon(
+            shell=[(0, 0), (0, 4), (4, 4), (4, 0)],
+            holes=[[(0, 1), (0, 5), (5, 5), (5, 1)]],
+        )
 
 
 def test_polygon_normalize() -> None:
```

Comment:
Should probably also add a test here with holes, to verify those also get normalized/oriented?